### PR TITLE
Fixed #31002 -- Fixed GIS lookups crash against a subquery annotation.

### PR DESCRIPTION
**Pull request by charettes**
https://code.djangoproject.com/ticket/31002

Thanks Vasileios Bouzas for the report.

### CODE REVIEW

File: django/contrib/gis/db/models/lookups.py [hunk 1]
```diff
@@ -75,9 +75,9 @@ def get_rhs_op(self, connection, rhs):
         return connection.ops.gis_operators[self.lookup_name]
 
     def as_sql(self, compiler, connection):
-        lhs_sql, sql_params = self.process_lhs(compiler, connection)
+        lhs_sql, lhs_params = self.process_lhs(compiler, connection)
         rhs_sql, rhs_params = self.process_rhs(compiler, connection)
-        sql_params.extend(rhs_params)
+        sql_params = tuple(lhs_params) + tuple(rhs_params)
```
Comment (sir-sigurd) [inline django/contrib/gis/db/models/lookups.py]:
```suggestion
        sql_params = (*a, *b)
```

Comment (sir-sigurd) [inline django/contrib/gis/db/models/lookups.py]:
```suggestion
        sql_params = (*lhs_params, *rhs_params)
```

**Comment by felixxm** [comment]
@charettes Thanks for this patch :+1: I pushed minor edits, and fixed `SDORelate` crash on Oracle.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Pull request merged**